app_feup/lib/controller/local_storage/app_print_movements_database.dart

```diff
@@ -0,0 +1,48 @@
+import 'dart:async';
+import 'package:uni/controller/local_storage/app_database.dart';
+import 'package:sqflite/sqflite.dart';
+
+/// Manages the app's Print Movements database.
+///
+/// This database stores information about the user's print movements
+class AppPrintMovementsDatabase extends AppDatabase {
+  AppPrintMovementsDatabase()
+      : super('print_movements.db', [
+          'CREATE TABLE movements(datetime DATETIME, value TEXT)',
+        ]);
+
+  Future<List> printMovements() async {
+    // Get a reference to the database
+    final Database db = await this.getDatabase();
+
+    // Query the table for all movements
+    final List movements = await db.query('movements');
+
+    return movements;
+  }
+
+  /// Adds all entries from [movements] to this database.
+  Future<void> _insertPrintMovements(List movements) async {
+    movements.forEach((movement) async {
+      await insertInDatabase(
+        'movements',
+        {'datetime': movement['datetime'], 'value': movement['value']},
+        conflictAlgorithm: ConflictAlgorithm.replace,
+      );
+    });
+  }
+
+  /// Deletes all of the movements from this database.
+  Future<void> deletePrintMovements() async {
+    // Get a reference to the database
+    final Database db = await this.getDatabase();
+    await db.delete('movements');
+  }
+
+  /// Replaces all the movements in this database with entries
+  /// from [movements].
+  Future<void> setPrintMovements(List movements) async {
+    await deletePrintMovements();
+    await _insertPrintMovements(movements);
+  }
```

Comment:
Can you make this a transaction?

Reply:
I think I done it in the right way, can you check please?

Reply:
Yes it's good, but i Think that we can remove the await in the insertPrintMovements, it doesn't seem to be necessary

Reply:
I was already talking about the changes in the new commit, from line 33 to 47. Sorry, my bad

```Dart
  /// Replaces all the movements in this database with entries
  /// from [movements].
  Future<void> setPrintMovements(List movements) async {
    final Database db = await this.getDatabase();

    await db.transaction((txn) async {
      await txn.delete('movements');

      movements.forEach((movement) async {
        await insertInDatabase(
          'movements',
          {'datetime': movement['datetime'], 'value': movement['value']},
          conflictAlgorithm: ConflictAlgorithm.replace,
        );
      });
    });
  }
```

Reply:
Mb also, I was talking about this 'await insertInDatabase(...'

Reply:
Consider it done!